lib/runner.js

```diff
@@ -314,6 +314,7 @@ Runner.prototype.hook = function (name, fn) {
             suite.tests.forEach(function (test) {
               test.pending = true;
             });
+            suite.parent.pending = true;
```

Comment:
I believe this will still skip the current suite's before/after hooks. It also might skip further tests and after hooks of the parent suite, but I'm not totally sure of that. 

You could try `suite.suites.forEach(function (suite) { suite.pending = true; })` instead.

Reply:
@leedm777 Thanks for the suggestion! I reworked my tests and realized that you are correct, any `after` hooks placed under the skipped `describe` were not executing. The only problem I'm having with your solution is when `this.skip` is called asynchronously. I updated my commit with the latest work.

From what I can tell, the asynchronous skip is aborting the test run, giving me outputs like this (there should be another skipped test in `inner inner`:

```
  outer
    ✓ should run
    inner
      - should not run
      inner inner


  1 passing (2s)
  1 pending
``` 

I feel like I'm missing something, so if someone with more experience in the code base has any insight I'd be glad to hear it.